chipflow_lib/platforms/sim.py

```diff
@@ -1,96 +1,145 @@
+
+        output = rtlil.convert(e, name="sim_top", ports=ports, platform=self)
 
         top_rtlil = Path(self.build_dir) / "sim_soc.il"
         with open(top_rtlil, "w") as rtlil_file:
             for box_content in self.sim_boxes.values():
                 rtlil_file.write(box_content)
             rtlil_file.write(output)
+
         top_ys = Path(self.build_dir) / "sim_soc.ys"
         with open(top_ys, "w") as yosys_file:
             for extra_filename, extra_content in self.extra_files.items():
                 extra_path = Path(self.build_dir) / extra_filename
                 with open(extra_path, "w") as extra_file:
                     extra_file.write(extra_content)
                 if extra_filename.endswith(".il"):
-                    print(f"read_rtlil {extra_filename}", file=yosys_file)
+                    print(f"read_rtlil {extra_path}", file=yosys_file)
                 else:
                     # FIXME: use -defer (workaround for YosysHQ/yosys#4059)
-                    print(f"read_verilog {extra_filename}", file=yosys_file)
+                    print(f"read_verilog {extra_path}", file=yosys_file)
             print("read_rtlil sim_soc.il", file=yosys_file)
             print("hierarchy -top sim_top", file=yosys_file)
-            # FIXME: use the default -O6 (workaround for YosysHQ/yosys#4227)
-            print("write_cxxrtl -O4 -header sim_soc.cc", file=yosys_file)
+            print("write_cxxrtl -header sim_soc.cc", file=yosys_file)
+
+    def instantiate_ports(self, m: Module):
+        if hasattr(self, "_pinlock"):
+            return
+
+        pinlock = load_pinlock()
+        for component, iface in pinlock.port_map.items():
+            for k, v in iface.items():
+                for name, port in v.items():
+                   invert = port.invert if port.invert else False
+                   self._ports[port.port_name] = io.SimulationPort(port.direction, port.width, invert=invert, name=f"{component}-{name}")
+
+        for clock, name in self._config["chipflow"]["clocks"].items():
+            if name not in pinlock.package.clocks:
+                raise ChipFlowError(f"Unable to find clock {name} in pinlock")
+
+            port_data = pinlock.package.clocks[name]
+            port = io.SimulationPort(io.Direction.Input, port_data.width, invert=True, name=f"clock-{name}")
+            self._ports[name] = port
+
+            if clock == 'default':
+                clock = 'sync'
+            setattr(m.domains, clock, ClockDomain(name=clock))
+            clk_buffer = io.Buffer("i", port)
+            setattr(m.submodules, "clk_buffer_" + clock, clk_buffer)
+            m.d.comb += ClockSignal().eq(clk_buffer.i)
+
+        for reset, name in self._config["chipflow"]["resets"].items():
+            port_data = pinlock.package.resets[name]
+            port = io.SimulationPort(io.Direction.Input, port_data.width, invert=port.invert, name=f"clock-{name}", )
```

Comment:
what is `port.invert` supposed to refer to? `port` isn't yet set at the time you use it, so it will actually be referring to the last clock or reset....

Reply:
random mess, i think, no real way to say if resets are inverted or not, so this should either be always not inverted or always inverted...